File: android-beans/src/main/java/rocks/frieler/android/beans/scopes/activity/ForegroundActivityHolder.java
package rocks.frieler.android.beans.scopes.activity;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

/**
 * Implementation of the {@link android.app.Application.ActivityLifecycleCallbacks} that holds the {@link Activity}
 * which is currently in the foreground (if any).
 */
public class ForegroundActivityHolder implements Application.ActivityLifecycleCallbacks {
    private Activity currentActivity = null;

    public Activity getCurrentActivity() {
        return currentActivity;
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle bundle) {
        this.currentActivity = activity;
    }

    @Override
    public void onActivityStarted(Activity activity) {
        this.currentActivity = activity;
    }

    @Override
    public void onActivityResumed(Activity activity) {
        this.currentActivity = activity;
    }

    @Override
    public void onActivityPaused(Activity activity) {
        if (this.currentActivity == activity) {
            this.currentActivity = null;
        }
    }

    @Override
    public void onActivityStopped(Activity activity) {
        if (this.currentActivity == activity) {
            this.currentActivity = null;
        }
    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle bundle) {}

    @Override
    public void onActivityDestroyed(Activity activity) {}
}

File: android-beans/src/main/kotlin/rocks/frieler/android/beans/scopes/activity/ForegroundActivityHolder.kt
package rocks.frieler.android.beans.scopes.activity

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

/**
 * Implementation of the [android.app.Application.ActivityLifecycleCallbacks] that holds the [Activity]
 * which is currently in the foreground (if any).
 */
class ForegroundActivityHolder : ActivityLifecycleCallbacks {
	var currentActivity: Activity? = null
		private set

	override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
		currentActivity = activity
	}

	override fun onActivityStarted(activity: Activity) {
		currentActivity = activity
	}

	override fun onActivityResumed(activity: Activity) {
		currentActivity = activity
	}

	override fun onActivityPaused(activity: Activity) {
		if (currentActivity === activity) {
			currentActivity = null
		}
	}

	override fun onActivityStopped(activity: Activity) {
		if (currentActivity === activity) {
			currentActivity = null
		}
	}

	override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) {}
	override fun onActivityDestroyed(activity: Activity) {}
}
